bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java

```diff
@@ -334,4 +355,52 @@ public void postUpdate(String channelId, State state) {
     public Collection<Class<? extends ThingHandlerService>> getServices() {
         return Collections.singleton(LGWebOSActions.class);
     }
+
+    @Override
+    public void thingDiscovered(DiscoveryService source, DiscoveryResult result) {
+        if (THING_TYPE_WEBOSTV.equals(result.getThingTypeUID())) {
+            Map<String, Object> properties = result.getProperties();
+            String host = (String) properties.get(CONFIG_HOST);
+            if (host != null && host.equals(getLGWebOSConfig().getHost())) {
+                // Thing matching the discovery
+                logger.debug("thingDiscovered: discovery with UID {} matching thing {}", result.getThingUID(),
+                        getThing().getUID());
+
+                channelHandlers.get(CHANNEL_POWER).onDeviceReady(CHANNEL_POWER, this);
```

Comment:
websocket connected iff power on

for users, which don't use upnp this wont work.
it may create a situation, when power channel is on, but we are not connected yet.

Reply:
I don't understand this part of your comment "for users, which don't use upnp this wont work.". The binding uses UPnP discovery, this is not a user's choice.
And for the few users having UPnP not working in their network, of course this PR will not help them as it is based on UPnP. But it probably concerns only a very small minority.

You are right, it can happen, at startup, that the channel is set to ON just before the websocket is connected and TV is considered as registered, as this is done in parallel by 2 different threads. 

But this is required to have a fast update of the channel when you power on the TV while the websocket was still alive and connected.
In fact, the goal of this PR is a fast update of the power channel (ON or OFF), not based on websocket connection status because the websocket remains connected during a certain time after the TV is switched to OFF.

But I agree that this channel represents whether the TV is ON or OFF, not whether the websocket is connected and TV is registered. This is an essential point.

Reply:
Considering the PR #7355, I think that state update to ON will become unnecessary.